toolbox/db/db_get.m

```diff
@@ -44,14 +44,90 @@
     end
 end
 
+st = dbstack;
 varargout = {};
 switch contextName
-    case 'Subject'
-        iSubject = args{1};
-        varargout{1} = sql_query(sqlConn, 'select', 'subject', '*', struct('Id', iSubject));
+%% ==== SUBJECT ====
+    % Usage : sSubject = db_get('Subject', SubjectIDs,       Fields, isRaw);
+    %         sSubject = db_get('Subject', SubjectFileNames, Fields, isRaw);
+    %         sSubject = db_get('Subject', CondQuery,        Fields, isRaw);
+    %         sSubject = db_get('Subject');
+    % If isRaw is set: force to return the real brainstormsubject description
+    % (ignoring wether it uses protocol's default anatomy or not)    
+    case 'Subject'      
+        % Parse inputs
+        iSubjects = args{1};
+        fields = '*';   
+        isRaw = 0;
+        templateStruct = db_template('Subject');
+                
+        if ischar(iSubjects)
+            iSubjects = {iSubjects};
+        elseif isstruct(iSubjects)
+            condQuery = args{1};           
+        end
+        
+        if length(args) > 1
+            fields = args{2};
+            if ischar(fields)
+                fields = {fields};
+            end
+            for i = 1 : length(fields)
+                resultStruct.(fields{i}) = templateStruct.(fields{i});
+            end
+        else
+            resultStruct = templateStruct;
+        end
+                       
+        % isRaw parameter
+        if length(args) > 2
+            isRaw = args{3};
+        end
         
+        % Input is SubjectIDs or SubjectFileNames
+        if ~isstruct(iSubjects)
+            nSubjects = length(iSubjects);
+            sSubjects = repmat(resultStruct, 1, nSubjects);
+            for i = 1:nSubjects
+                if iscell(iSubjects)
+                    condQuery.FileName = iSubjects{i};
+                else
+                    condQuery.Id = iSubjects(i);
+                end
+                sSubjects(i) = sql_query(sqlConn, 'select', 'subject', fields, condQuery);
+            end
+        else % Input is struct query
+            sSubjects = sql_query(sqlConn, 'select', 'subject', fields, condQuery(1));
+        end
+
+        % Retrive default subject is needed
```

Comment:
Typos: Retrieve default subject if needed.

Reply:
Solved in 656d5cc318907a28da0d0c5823e1989782eb4ace